### ISSUE TITLE
Problem with setElementModfierValue and modifiers of replaceable classes

### ISSUE THREAD

**Issue by abuntrock** [OpenModelica/OpenModelica#12769]
### Description

`setElementModifierValue` is not able to set submodifiers of replaceable classes and instead deletes the complete redeclare statement in the process.

### Steps to Reproduce

I changed the example from #10190 a little. 

Run the following script: 
```
loadString("within ;
package TestOMEdit
  model ClasswithReplaceable

    replaceable TestOMEdit.Internal.ClassA testClass annotation(choices(
      choice(redeclare TestOMEdit.Internal.ClassA testClass \"A\"),
      choice(redeclare TestOMEdit.Internal.ClassB testClass \"B\")));
    replaceable parameter TestOMEdit.Internal.recordA testrecrod;

    annotation (Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
          coordinateSystem(preserveAspectRatio=false)));
  end ClasswithReplaceable;

  package Internal
    model ClassA
      parameter Real A = 1;
      annotation (Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
            coordinateSystem(preserveAspectRatio=false)));
    end ClassA;

    model ClassB
      parameter Real B = 2;
      annotation (Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
            coordinateSystem(preserveAspectRatio=false)));
    end ClassB;

    record recordA
      parameter Real a = 1;
      annotation (Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
            coordinateSystem(preserveAspectRatio=false)));
    end recordA;

    record recordB
      parameter Real B = 1;
      annotation (Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
            coordinateSystem(preserveAspectRatio=false)));
    end recordB;
  end Internal;
end TestOMEdit;

model M
  TestOMEdit.ClasswithReplaceable classwithReplaceable annotation(
    Placement(visible = true, transformation(origin = {-56, 24}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
end M;");
getErrorString();
getClassNames();
getErrorString();

setElementModifierValue(M, classwithReplaceable, $Code((redeclare TestOMEdit.Internal.ClassB testClass "B")));
getErrorString();
list(M);
getErrorString();

setElementModifierValue(M, classwithReplaceable.testClass.B, $Code(=100));
getErrorString();
list(M);
getErrorString();
```

The Output is: 
```
true
""
{TestOMEdit,M}
""
true
""
"model M
  TestOMEdit.ClasswithReplaceable classwithReplaceable(redeclare TestOMEdit.Internal.ClassB testClass \"B\") annotation(
    Placement(visible = true, transformation(origin = {-56, 24}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
end M;"
""
true
""
"model M
  TestOMEdit.ClasswithReplaceable classwithReplaceable annotation(
    Placement(visible = true, transformation(origin = {-56, 24}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
end M;"
""
```

### Expected Behavior

The first difference to #10190 (not aim of this issue) is that I have to use 
`setElementModifierValue(M, classwithReplaceable, $Code((redeclare TestOMEdit.Internal.ClassB testClass "B")));`
instead of 
`setElementModifierValue(M, classwithReplaceable.testClass, $Code((redeclare TestOMEdit.Internal.ClassB testClass "A")));`
to get the desired output. Was that an intentional change since this last ticket was closed? 

The desired output at the end is 
```
"model M
  TestOMEdit.ClasswithReplaceable classwithReplaceable(redeclare TestOMEdit.Internal.ClassB testClass(B = 100) \"B\") annotation(
    Placement(visible = true, transformation(origin = {-56, 24}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
end M;"
```
which can be achieved by combining the two commands into the following one:
`setElementModifierValue(M, classwithReplaceable, $Code((redeclare TestOMEdit.Internal.ClassB testClass(B=100) "B")))`.
However, this is not helpful when dealing with multiple modifiers of a replaceable class. When the class, is not replaceable submodifiers at any arbitrary level can be successfully changed with `setElementModifierValue`.

### Version and OS

 - OpenModelica Version: 1.23.0
 - OS: Windows 11, 64 bit

Thanks for looking into this.
Aaron Buntrock - Bosch Rexroth

**Comment by casella** [comment]
@perost please check this at your earliest convenience. I think the fix could go into 1.23.2

**Comment by perost** [comment]
Should be fixed with #12795. I can port it to the maintenance branch later, but I'm not completely confident that it doesn't break anything so I'd prefer to give it time to get some practical testing.

>The first difference to https://github.com/OpenModelica/OpenModelica/issues/10190 (not aim of this issue) is that I have to use
>setElementModifierValue(M, classwithReplaceable, $Code((redeclare TestOMEdit.Internal.ClassB testClass "B")));
>instead of
>setElementModifierValue(M, classwithReplaceable.testClass, $Code((redeclare TestOMEdit.Internal.ClassB testClass "A")));
>to get the desired output. Was that an intentional change since this last ticket was closed?

The current behaviour is the intended one, the second argument should be the name of the element that the modifier will be placed on. It was changed by #10616, before that you had to add the name of the redeclare element to the path which was inconsistent with normal modifiers. That wasn't really intended and was making it hard to handle redeclares and non-redeclares in the same call, so it had to be changed to be more consistent.

**Comment by abuntrock** [comment]
Thanks. I will test this with tomorrow's nightly.

And thanks for the explanation. I also think that the new approach is much more intuitive.

**Comment by abuntrock** [comment]
Looks perfect with today's nightly. Thanks a lot

**Comment by casella** [comment]
@abuntrock please come back to us in a few weeks after some testing if you are interested at getting this fix in version 1.23.2

**Comment by abuntrock** [comment]
@casella This works fine for us. We would be glad if this fix could make its way into 1.23.2

**Comment by casella** [comment]
@abuntrock it will be in 1.24.0 which is going to be released soon, a beta will be probably available next week.
